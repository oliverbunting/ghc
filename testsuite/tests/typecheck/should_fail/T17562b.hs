{-# LANGUAGE QuantifiedConstraints, MultiParamTypeClasses,
             PolyKinds #-}

module T17562b where

class (forall a. a b ~ a c) => C b c
