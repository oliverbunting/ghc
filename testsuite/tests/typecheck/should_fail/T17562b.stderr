
T17562b.hs:6:1: error:
    • Unconstrained kind variable k1 in superclass context:
        class (forall (a :: k -> k1). (a b :: k1) ~ (a c :: k1)) =>
              C @{k} (b :: k) (c :: k)
      Perhaps add a kind signature to a local type variable.
    • In the class declaration for ‘C’
